test/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.test.cpp

```diff
@@ -400,6 +400,66 @@ TEST_F(OpenSpaceToolkit_Astrodynamics_Trajectory_Orbit_Models_Propagator, Calcul
     {
         EXPECT_NO_THROW(defaultPropagator_.calculateStateAt(state, instant));
     }
+
+    {
+        // Test that the propagator respects the input Frame
+        const State outputStateGCRF = defaultPropagator_.calculateStateAt(state, instantArray[0]);
+        const State outputStateITRF =
+            defaultPropagator_.calculateStateAt(state.inFrame(Frame::ITRF()), instantArray[0]);
+
+        EXPECT_EQ(outputStateGCRF.getFrame(), Frame::GCRF());
+        EXPECT_EQ(outputStateITRF.getFrame(), Frame::ITRF());
+
+        const State outputStateGCRF2 = outputStateITRF.inFrame(Frame::GCRF());
+
+        EXPECT_EQ(outputStateGCRF.getInstant(), outputStateGCRF2.getInstant());
+        EXPECT_EQ(outputStateGCRF.getFrame(), outputStateGCRF2.getFrame());
+        EXPECT_EQ(outputStateGCRF.accessCoordinatesBroker(), outputStateGCRF2.accessCoordinatesBroker());
+
+        // Get the coordinates from the two states
+        const VectorXd coordinates1 = outputStateGCRF.getCoordinates();
+        const VectorXd coordinates2 = outputStateGCRF2.getCoordinates();
+
+        // Check that the two vectors have the same size
+        EXPECT_EQ(coordinates1.size(), coordinates2.size());
+
+        // Check that each element of the two vectors is close
+        for (int i = 0; i < coordinates1.size(); ++i)
+        {
+            EXPECT_NEAR(coordinates1(i), coordinates2(i), 1e-5);
```

Comment:
I feel like it should be a lot closer than 1e-5, was this the best tolerance you got?

Reply:
Nah, this isn't really testing to any meaningful tolerance. The point here was just to ensure that the transformation is actually happening. If it weren't, I'd expect errors on the order of meters or kilometers. I can add a comment to that effect for clarity